### PR TITLE
[feat] add right and left jacobian for So3

### PR DESCRIPTION
**Pull request by cjpurackal**
#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/geometry/conversions.py [hunk 1]
```diff
@@ -1478,3 +1479,35 @@ def ARKitQTVecs_to_ColmapQTVecs(qvec: Tensor, tvec: Tensor) -> tuple[Tensor, Ten
...
+        >>> vector_to_skew_symmetric_matrix(vec)
+        tensor([[ 0., -3.,  2.],
+                [ 3.,  0., -1.],
+                [-2.,  1.,  0.]])
+    """
+    # KORNIA_CHECK_SHAPE(vec, ["B", "3"])
+    if vec.shape[-1] != 3 or len(vec.shape) > 2:
+        raise ValueError(f"Input vector must be of shape (B, 3) or (3,). " f"Got {vec.shape}")
+    x1, x2, x3 = vec[..., 0], vec[..., 1], vec[..., 2]
+    zeros = torch.zeros_like(x1)
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:1504]:
```suggestion
    zeros = zeros_like(x1)
```
use from kornia.core to later migrate easily

File: kornia/geometry/conversions.py [hunk 2]
```diff
@@ -1478,3 +1479,35 @@ def ARKitQTVecs_to_ColmapQTVecs(qvec: Tensor, tvec: Tensor) -> tuple[Tensor, Ten
     t_colmap = t_colmap.reshape(-1, 3, 1)
     q_colmap = rotation_matrix_to_quaternion(R_colmap.contiguous())
     return q_colmap, t_colmap
+
+
+def vector_to_skew_symmetric_matrix(vec: Tensor) -> Tensor:
+    r"""Converts a vector to a skew symmetric matrix.
```
Comment (edgarriba) [inline kornia/geometry/conversions.py:1485]: can you add some formulas ? https://math.stackexchange.com/questions/258775/from-a-vector-to-a-skew-symmetric-matrix

**Review by edgarriba — APPROVED** [review]

File: kornia/geometry/conversions.py (hunk 1 above)
Comment (cjpurackal) [reply kornia/geometry/conversions.py:1504]: we should probably do that for the rest of the functions there as well

**Review by johnnv1 — APPROVED** [review]

File: kornia/geometry/liegroup/so3.py [hunk 3]
```diff
@@ -331,3 +332,63 @@ def adjoint(self) -> Tensor:
...
+            >>> vec = torch.tensor([1., 2., 3.])
+            >>> So3.left_jacobian(vec)
+            tensor([[-0.0687, -0.2267,  0.5074],
+                    [ 0.5556,  0.1779,  0.3629],
+                    [-0.0141,  0.6236,  0.5890]])
+        """
+        # KORNIA_CHECK_SHAPE(vec, ["B", "3"])  # FIXME: resolve shape bugs. @edgarriba
+        R_skew = vector_to_skew_symmetric_matrix(vec)
+        theta = vec.norm(dim=-1, keepdim=True)[..., None]
+        I = eye(3, device=vec.device, dtype=vec.dtype)
```
Comment (johnnv1) [inline kornia/geometry/liegroup/so3.py:383]: `E741 Ambiguous variable name` from flake8 -- https://www.flake8rules.com/rules/E741.html

Comment (edgarriba) [reply kornia/geometry/liegroup/so3.py:383]: i think we should find a trade-off with mathematical notation or a way to detect/skip

Comment (johnnv1) [reply kornia/geometry/liegroup/so3.py:383]: i don't know if is possible to have a supplementary dict to allow some cases -- but we always can `# noqa: E741` on each line

**Pull request merged**